google/cloud/spark_connect/session.py

```diff
@@ -524,3 +505,43 @@ def stop(self) -> None:
                 GoogleSparkSession._active_session, "session", None
             ):
                 GoogleSparkSession._active_session.session = None
+
+
+class ServerlessSessionHelper:
+
+    @staticmethod
+    def terminate_s8s_session(
+        project_id, region, active_s8s_session_id, client_options=None
+    ):
+        from google.cloud.dataproc_v1 import SessionControllerClient
+
+        logger.debug(f"Terminating serverless session: {active_s8s_session_id}")
+        terminate_session_request = TerminateSessionRequest()
+        session_name = f"projects/{project_id}/locations/{region}/sessions/{active_s8s_session_id}"
+        terminate_session_request.name = session_name
+        state = None
+        try:
+            SessionControllerClient(
+                client_options=client_options
+            ).terminate_session(terminate_session_request)
+            get_session_request = GetSessionRequest()
+            get_session_request.name = session_name
+            state = Session.State.ACTIVE
+            while (
+                state != Session.State.TERMINATING
+                and state != Session.State.TERMINATED
+                and state != Session.State.FAILED
+            ):
+                session = SessionControllerClient(
+                    client_options=client_options
+                ).get_session(get_session_request)
+                state = session.state
+                sleep(1)
+        except NotFound:
+            logger.debug(f"Session {active_s8s_session_id} already deleted")
+        except FailedPrecondition:
+            logger.debug(
+                f"Session {active_s8s_session_id} already terminated manually or terminated automatically through session ttl limits"
+            )
+        if state is not None and state == Session.State.FAILED:
+            raise RuntimeError("Serverless session termination failed")
```

Comment:
Instead of this logic with the `state = None` statement above, put a `return` statement outside of the while loop on line 540, and then make the subsequent `raise RuntimeError...` call below unconditional.